src/watchmaker/utils/__init__.py

```diff
@@ -119,3 +121,34 @@ def clean_none(value):
         return None
 
     return value
+
+
+def set_file_perms(path, dir_mode=None, file_mode=None):
+    r"""
+    Set all file and directory permissions at or under path to modes.
+
+    Args:
+        path: (:obj:`str`)
+            Directory to be secured.
+
+        dir_mode: (`int)
+            Mode to be applied to directories.
+
+        file_mode: (`int`)
+            Mode to be applied to files.
+
+    """
+    if not dir_mode:
+        dir_mode = stat.S_IEXEC | stat.S_IWRITE | stat.S_IREAD
+
+    if not file_mode:
+        file_mode = stat.S_IWRITE | stat.S_IREAD
+
+    for root, dirs, files in os.walk(path, topdown=False):
+        for one_file in [os.path.join(root, f) for f in files]:
```

Comment:
it seems a bit excessive to perform the join here. this is a loop in a loop in a loop. move the join to the call to `oschmod.set_mode` and it's just a loop in a loop...
```suggestion
        for one_file in files:
            oschmod.set_mode(os.path.join(root, one_file), file_mode)
```

same for the loop over `dirs`...